graylog2-web-interface/src/util/FileDownloadUtils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { fetchFile } from 'logic/rest/FetchProvider';
+
+export const downloadBLOB = (contents: BlobPart, metadata: { fileName: string, contentType: string }) => {
+  // create blob from contents and meta data
+  const blob = new Blob([contents], { type: metadata.contentType });
+
+  // eslint-disable-next-line compat/compat
+  const href = window.URL.createObjectURL(blob);
+  const a = document.createElement('a');
+  a.href = href;
+  a.download = metadata.fileName;
+
+  if (document.body) {
+    document.body.appendChild(a);
+  }
+
+  // perform download
+  a.click();
+  a.remove();
+};
+
```

Comment:
It's not the same. For messages, we get the file name and then download it from the server. In my case, we use a blob to download it.

Reply:
It seems to be the exact same code like in `UploadFieldStore`? For me it event looks like you copied it because event the comment is the same!? I mainly just want to avoid the redundant code.

Regarding `MessageExportUtils` you are right we do not use a blob here, but we could still create a shared helper function for adding the anchor the the DOM and clicking on it.

Reply:
`UploadFieldStore`  from an enterprise? Sorry, I tried to find `UploadFieldStore` in PR 10617 and I didn't and thought that it was some old thing that doesn't exist anymore. In enterprise, I planned to replace it after the merging of this PR.

Reply:
Added shared function to create `<a> `and download